ops/charm.py

```diff
@@ -1488,6 +1560,90 @@ def __init__(self, name: str, raw: Dict[str, Any]):
         self.type = raw['type']
 
 
+@dataclasses.dataclass(frozen=True)
+class DeviceMeta:
```

Comment:
Do we have any charms that use `Device`? This was something added way back in the day for the older style charms. If there's no sidecar charms using this, I would recommend not implementing it until it's asked for. That way we don't lock ourselves into supporting something that turns out to be "wrong".

Reply:
In the ~150 I have a list of, there are none. @benhoyt given that this is an 'old' thing not a 'new' one, I'm happy to remove it - fine with you?

Reply:
Yeah, Ian's rationale makes sense -- let's remove it from the PR.